test/testutil/provider.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018-2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "../testutil.h"
```

Comment:
you might want to see what app_create_libctx() does instead..
this gets enabled if export OPENSSL_TEST_LIBCTX="1"

I have used -config in other apps combined with -provider.